onnxruntime/core/providers/cuda/tensor/where.cc

```diff
@@ -114,21 +114,21 @@ struct TernaryElementwisePreparation {
       // to decide if dim0 needs broadcast
       a_padded_strides.AllocCpuPtr(out_rank + 1);
       ORT_RETURN_IF_NOT(TensorPitches::Calculate(a_padded_strides.CpuSpan(), a_shape.GetDims()));
-      if (a_shape[0] > 1 && a_rank == out_rank)
+      if (a_rank > 0 && a_shape[0] > 1 && a_rank == out_rank)
```

Comment:
can we return earlier once we found a_rank == 0?

Reply:
a_rank == 0 is a valid case and we still need to set the values in a_padded_strides. We just no need to change a_padded_strides[0] when a_rank == 0, so I think we should not return.